dnnv/verifiers/common/reductions/iopolytope/base.py

```diff
@@ -103,8 +103,12 @@ def __init__(self, variable=None):
         self._b: List[np.ndarray] = []
         self._A_mat: Optional[np.ndarray] = None
         self._b_vec: Optional[np.ndarray] = None
-        self._lower_bound = np.full(self.size(), -np.inf)
-        self._upper_bound = np.full(self.size(), np.inf)
+        if np.isscalar(self.size()):
+            shape = self.size()
+        else:
+            shape = self.size()[::-1]
+        self._lower_bound = np.full(shape, -np.inf)
+        self._upper_bound = np.full(shape, np.inf)
```

Comment:
This does not seem to address the issue specified. Constraint.size() will always return a scalar value so only one branch of the conditional will ever be taken, and that branch leads to behavior that is equivalent to the changed code.

Reply:
Hi,
I'm so sorry.
I'm not entirely sure how this happened, but for some reason the important part of the fix wasn't committed yesterday.
I've now restructured the fix:
- This if branch is no longer used (I now realize that it was useless)
- The two invocations of `linprog` which previously had bounds `(None,None)` now use the previously obtained bounds (this was the part that was missing yesterday).

I've just rerun the tests locally and this should, hopefully fix this bug for good.

Reply:
Ah, that makes sense. I think I used to include the bounds in the A and b matrices, but at some point that was refactored out and I must have missed updating that here.